src/helpers/repoutils.ts

```diff
@@ -4,22 +4,29 @@
 *--------------------------------------------------------------------------------------------*/
 "use strict";
 
+import { Logger } from "../helpers/logger";
+import * as url from "url";
+
 //This class is responsible for determining information about a string-based url (not a uri)
 //A Team Services Git repository is determined by:
-// "_git" AND ".visualstudio.com" in the url
+// "_git" AND ".visualstudio.com" / "azure.com" in the url
 //A TFS Git repository is determined by:
-// "_git" in the url and ".visualstudio.com" NOT in the url
+// "_git" in the url and ".visualstudio.com" / "azure.com" NOT in the url
 //A Team Services TFVC repository is determined by:
-// "_git" NOT in the URL and ".visualstudio.com" in the url
+// "_git" NOT in the URL and ".visualstudio.com" / "azure.com" in the url
+//A Team Services azure repository is determined by:
+// "azure.com" in the hostname
 //A TFS TFVC repository is determined by:
-// "_git" NOT in the URL and ".visualstudio.com" NOT in the url
+// "_git" NOT in the URL and ".visualstudio.com" / "azure.com" NOT in the url
 //
 //NOTE: the detection of a TFS TFVC repository is basically a "catch all".  Any url could match
 //it and we'd think it's a TFVC repo.  For that to be true, this class now assumes that it is
 //always detecting either a Team Services or TFS repository.  It no longer tries to determine
 //that a url is NOT either a TFS or Team Services repository.  So it is up to the caller to only
 //instantiate this class if we know we should have either a TFS or Team Services repository.
 export class RepoUtils {
+    private static sshV3 = new RegExp("git@(?:vs-)?ssh\.(.+):v3\/(.+)\/(.+)\/(.+)");
```

Comment:
VSTS now introduced a new version where the `git@` is now replaced with the `<VS account name>@`.

![image](https://user-images.githubusercontent.com/5327862/45491750-24200380-b773-11e8-9972-73cdd1cf91bb.png)

Which means, the experience for SSH-cloned repos is broken again.

One thing to note, the old URIs, starting with `git@` still work -- however, since the VSTS experience exposes the new format, then that's what people are going to use.

Reply:
Thank You @johnib !  If you're interested in fixing this please submit a PR and we'll happily review.